### DISCUSSION TITLE
How to save the rendered image in android?

### DISCUSSION THREAD

**Discussion by ViciousDove19** [google/filament#5073] [Q&A] [answered]
I have no experience in using android studio or Kotlin. I am using filament to generate a dataset for a deep learning project. I was able to get the renderer working but I don't know how to save the rendered image. I don't even mind saving the screenshot but I could not get it to work through code.

For context, what I am trying to do is to generate images for different lighting conditions. I wonder if SwapChain is useful but have no idea how to use it. Any help is appreciated.

**Comment by romainguy** [comment, marked as answer]
The easiest way on Android is to use [PixelCopy](https://developer.android.com/reference/android/view/PixelCopy). You can give it either the `Surface` you are rendering into (better) or the `SurfaceView`.

**Reply by ViciousDove19** [reply]
Yes! That's what I did and it worked. Thank you!
